### ISSUE TITLE
btrbk 0.19.1 dies when using btrfs-progs < 3.17.3 and target subvolume is not mounted

### ISSUE THREAD

**Issue by bk1285** [digint/btrbk#24]
Hi!

A dryrun with this config file

    snapshot_dir               btrbk_snapshots
    incremental                yes
    snapshot_create always
    resume_missing             yes
    preserve_day_of_week       monday

    snapshot_preserve_daily    14
    snapshot_preserve_weekly   8
    snapshot_preserve_monthly  all

    target_preserve_daily      14
    target_preserve_weekly     8
    target_preserve_monthly    all

    btrfs_commit_delete        after

    volume /home
      subvolume user1
        target send-receive    /mnt/btrfs_bkp

fails like this

    btrbk command line client, version 0.19.1  (Sat Jun  6 11:09:09 2015)
    Using configuration: /foo/bar/btrbk.conf

    ERROR: process died unexpectedly (btrbk v0.19.1)
    Please contact the author: Axel Burri <axel@tty0.ch>

    Stack Trace:
    ----------------------------------------
    Use of uninitialized value $key in concatenation (.) or string at /usr/sbin/btrbk line 367, <FILE> line 95.
     at /usr/sbin/btrbk line 119, <FILE> line 95.
    	main::__ANON__("Use of uninitialized value \$key in concatenation (.) or strin"...) called at     /usr/sbin/btrbk line 367
    	main::check_file("ERROR", HASH(0x1c49170)) called at /usr/sbin/btrbk line 632
	main::btrfs_subvolume_detail(HASH(0x1b8bf08)) called at /usr/sbin/btrbk line 291
	main::vinfo_root(HASH(0x1b8bf08)) called at /usr/sbin/btrbk line 1728

Do you have any hint?

**Comment by digint** [comment]
Thanks for the bug report!

At some point, `btrfs subvolume show` returns an error, which is not correctly parsed by btrbk (causing the above error).

If you run btrbk with `-l debug`, you will be able to see what command fails exactly. Could you please paste the debug output here? I'll dig into this later.

**Comment by bk1285** [comment]
Hi, here the output

btrbk command line client, version 0.19.1  (Sat Jun  6 16:21:30 2015)
Using configuration: /foo/bar/btrbk.conf
### btrfs subvolume show '/home' 2>&1
Command execution successful
Parsed 11 subvolume detail items: /home
### btrfs subvolume list -a -c -u -q -R '/home'
Command execution successful
Parsed 39 total subvolumes for filesystem at: /home
Found 4 subvolume children of: /home
### btrfs subvolume show '/mnt/btrfs_bkp' 2>&1
Command execution successful
Real path for subvolume "/mnt/btrfs_bkp" is: ERROR

ERROR: process died unexpectedly (btrbk v0.19.1)
Please contact the author: Axel Burri <axel@tty0.ch>

Stack Trace:
----------------------------------------
Use of uninitialized value $key in concatenation (.) or string at /usr/sbin/btrbk line 367, <FILE> line 95.
 at /usr/sbin/btrbk line 119, <FILE> line 95.
	main::__ANON__("Use of uninitialized value \$key in concatenation (.) or strin"...) called at /usr/sbin/btrbk line 367
	main::check_file("ERROR", HASH(0x19e4350)) called at /usr/sbin/btrbk line 632
	main::btrfs_subvolume_detail(HASH(0x1941280)) called at /usr/sbin/btrbk line 291
	main::vinfo_root(HASH(0x1941280)) called at /usr/sbin/btrbk line 1728


/mnt/btrfs_bkp is an empty directory (= mountpoint for my usb-drive, whenever it is attached. Currently it isn't...)
Since the top-level directory ('/') is a btrfs-volume as well, /mnt/btrfs_bkp is a subdirectory within a btrfs-device. Might this be an issue?

**Comment by digint** [comment]
I traced the error down to btrfs-progs < 3.17.3, which returns exit status 0 on errors:
```
commit 05afee34683f5a57fba0b2f55375bef592bb3987
btrfs-progs: fix return value problem for btrfs sub show
```
This totally confuses btrbk ("Command execution successful") and leads to your error.

Fixed in: 16cfb600b72ac73e6e3ed5946fd358b6384c6637

**Comment by digint** [comment]
@bk1285 please confirm that you are using btrfs-progs < 3.17.3 and btrbk-0.19.2-dev (on master branch) fixes the problem.

**Comment by bk1285** [comment]
Great! Works now for me with btrbk-0.19.2-dev and btrfs-progs version 3.17!

Regards,
Bernd

**Comment by digint** [comment]
fix included in btrbk-0.19.2
